benchmark_runner/main/main.py

```diff
@@ -151,11 +155,15 @@ def run_benchmark_runner_workload():
         install_resources()
     elif ci_status == 'pass' or ci_status == 'failed':
         update_ci_status()
-    elif environment_variables.get_workload_namespace(workload) == 'benchmark-operator':
+    elif is_benchmark_operator_workload:
         run_benchmark_operator_workload()
-    elif environment_variables.get_workload_namespace(workload) == 'benchmark-runner':
-        run_benchmark_runner_workload()
 
+    elif is_benchmark_runner_workload:
+        success = run_benchmark_runner_workload()
+        if not success: sys.exit(1)
```

Comment:
Please raise custom error, its not recommend to use sys.exit

Reply:
this is the `main` function, and it's not called with `sys.exit(main())` (otherwise, I would have replaced `sys.exit(n))`  by `return n`)
it's not recommended to throw exception out of the `main()` function, it should rather return a proper error code

what about this then?
```
if not success: return 1
```
and
```
sys.exit(main())
```

Reply:
Agreed.  I hate seeing stack traces out of every error.

Reply:
The right way is to raise a custom Exception

Reply:
@ebattat you can't ignore my comment and not give any source for what you ask for ....

https://docs.python.org/3/library/sys.html#sys.exit
```
sys.exit([arg])
    Raise a SystemExit exception, signaling an intention to exit the interpreter.
```
I don't see anything against using it from the `main()` function (nor anywhere else in fact).

Raising a _custom_ exception in the `main()` is wrong, we need a way to define the return code of the process, which is the standard Unix convention to define if the process succeeded / why it failed.
```
raise SystemExit(N)
```
is another alternative, also it's semantically identical to the current code.